### PR TITLE
Add the packaging metadata to build the iroha snap

### PR DESCRIPTION
**Pull request by elopio**
### Description of the Change

This package will let you publish the latest iroha binaries in the Ubuntu store, and from there reach many users on all the supported Ubuntu versions, and Linux distributions. You just have to go to https://build.snapcraft.io and enable the automated continuous delivery. 

### Benefits
1-command installs, complete isolation when the snap is made strict, automatic updates, 4 channels with different levels of risks to make things easy and fun for your testers.

### Possible Drawbacks 

At this point, none. You can just give it a try in edge, and see if you like it before publishing it to stable. Once it's in stable, you will get users and will have to keep moving the releases from the edge channel when they are ready.

### CODE REVIEW

**Comment by neewy** [comment]
Can you change the target to be develop?

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    install: |
+      mkdir -p $SNAPCRAFT_PART_INSTALL/opt/sonar
+      mv ../src/build-wrapper-linux-x86 $SNAPCRAFT_PART_INSTALL/opt/sonar/build-wrapper
+      mkdir -p $SNAPCRAFT_PART_INSTALL/usr/local/bin/
+      ln -s -f ../../../opt/sonar/build-wrapper/build-wrapper-linux-x86-64 $SNAPCRAFT_PART_INSTALL/usr/local/bin/sonar-build-wrapper
+  valgrind:
+    source: git://sourceware.org/git/valgrind.git
+    plugin: autotools
+  ed25519:
+    source: https://github.com/warchant/ed25519.git
```
Comment (neewy) [inline snap/snapcraft.yaml]: can you please change it to be hyperledger/iroha-ed25519?

**Comment by elopio** [comment]
@neewy done and done. Thanks for taking a look!

**Comment by Warchant** [comment]
@elopio Thank you for your contribution. 

However, we can not setup snapcraft for Iroha, it prints this error:

image

Is it possible that someone else owns name `iroha` and therefore it does not want to register this name?

**Comment by elopio** [comment]
Hey @Warchant! The name iroha in the store was registered by an email address that starts with
`warchant`. Was that you? No revision has been pushed on that account, so I can transfer it to you without problems. You just have to click that link that says "file a claim" and fill the form.

**Comment by Warchant** [comment]
@elopio This is interesting! Now I see in my profile `hyperledger/iroha`, however when I tried to add it, the checkbox was grey (I am not hyperledger admin, so I don't have enough permissions to add repo).

I will ask @ryjones to fill the form for claim.

**Comment by ryjones** [comment]
I can't authenticate with it right now - I keep getting a server error. I will try later.

**Comment by elopio** [comment]
That's very weird, it works for me.

If you keep getting errors, can you please report it on https://forum.snapcraft.io/ ?

We are all traveling today and tomorrow for a big company sprint, so it might take some time to reply. But, the good thing is that starting on monday we'll all be together, so any issue you still have will take no time to get it solved. It would be awesome if you start releasing next week, we could even talk to our marketing team there to highlight your project in the store, our website and our social media.

**Comment by l4l** [comment]
DCO seems broken

**Comment by Warchant** [comment]
@l4l it is buggy: 
when you add `Leo Arias <leo.arias@canonical.com>`, it expects `elopio <yo@elopio.net>`, and vice versa.

**Comment by ryjones** [comment]
I'm getting that internal server error. I'll try again later.

**Comment by ryjones** [comment]
@elopio @Warchant I have filed a claim for the iroha name.

> Name
> iroha
> Private
> no
> Created
> Fri 02 March 2018 23:03:36
> State
> Pending name dispute review

**Comment by ryjones** [comment]
@Warchant @elopio It shows up in my account. It looks like I'm waiting for this change to be merged for the next step?

**Comment by elopio** [comment]
@ryjones yes. Once the snapcraft.yaml is on the branch, build.snapcraft.io will be able to build and push to edge on every change.

**Comment by Warchant** [comment]
Webhooks work, build works, PR is ready to be merged.

**Pull request merged**